src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -99,8 +99,12 @@ class Definitions {
   lazy val RootPackage: TermSymbol = ctx.newSymbol(
     NoSymbol, nme.ROOTPKG, PackageCreationFlags, TypeRef(NoPrefix, RootClass))
 
-  lazy val EmptyPackageClass = ctx.newCompletePackageSymbol(RootClass, nme.EMPTY_PACKAGE).moduleClass.asClass
-  lazy val EmptyPackageVal = EmptyPackageClass.sourceModule.entered
+  lazy val EmptyPackageVal = ctx.newCompletePackageSymbol(RootClass, nme.EMPTY_PACKAGE).entered
+  lazy val EmptyPackageClass = EmptyPackageVal.moduleClass.asClass
+
+  /** A package in which we can place all methods that are interpreted specially by the compiler */
```

Comment:
`throw` doesn't have to be intercepted, this could just go to JS, which will just throw whataver was the argument. If optimizer is not present, that's just a method call. If it's present - you can intensify such calls. It's not *required* to intensify such calls - as if compiled as calls to either JS or Java defined method they will perfectly run. 

I'm still not getting why `throw` *has* to be intrensified - it could just be some method in `scala.runtime.*` written in java, or javascript. But I totally agree about arrays.